### PR TITLE
Fix JournalDB era marker

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: util/src/journaldb/earlymergedb.rs [hunk 1]
```diff
@@ -435,7 +441,10 @@ impl JournalDB for EarlyMergeDB {
 				trace!(target: "jdb.ops", "  Deletes: {:?}", removes);
 			}
 			try!(batch.put(&last, r.as_raw()));
-			try!(batch.put(&LATEST_ERA_KEY, &encode(&now)));
+			if now >= self.latest_era {
```
Comment (gavofyork) [inline util/src/journaldb/earlymergedb.rs]: not just `>`?

**Comment by gavofyork** [comment]
looks good other than the question.

File: util/src/journaldb/earlymergedb.rs (hunk 1 above)
Comment (arkpar) [inline util/src/journaldb/earlymergedb.rs]: Can't use `0` as special value with `>`. Changed to `Option`

**Comment by gavofyork** [comment]
weren't there tests failing with this? do they pass now?

**Comment by gavofyork** [comment]
tests todo.

**Comment by arkpar** [comment]
Added a test

File: util/src/journaldb/overlayrecentdb.rs [hunk 2]
```diff
@@ -870,4 +870,24 @@ mod tests {
...
+		assert!(jdb.can_reconstruct_refs());
+
+		let bar = jdb.insert(b"bar");
+		jdb.commit(1, &b"1".sha3(), Some((0, b"0a".sha3()))).unwrap();
+		assert!(jdb.can_reconstruct_refs());
+
+		jdb.remove(&bar);
+		jdb.commit(0, &b"0b".sha3(), None).unwrap();
+		assert!(jdb.can_reconstruct_refs());
+		jdb.commit(2, &b"1".sha3(), Some((1, b"1".sha3()))).unwrap();
```
Comment (gavofyork) [inline util/src/journaldb/overlayrecentdb.rs]: should be `commit(2, &b"2".sha3(), ...`

**Pull request merged**